### PR TITLE
RFC: Linux build fixes

### PR DESCRIPTION
**Pull request by kylex2**
Hi,
I made a series of patches which fixes a bunch of warnings and enables cross compiling with the generic Makefile:

`no-OS/projects/ad9361$ TINYIIOD="y" PLATFORM="linux" make`

Only warnings left after this series is a couple of
`warning: #warning "TCP socket communication is not secured"`

Only build path tested is projects/ad9361 for Linux.

### CODE REVIEW

File: projects/ad9361/src/parameters.h [hunk 1]
```diff
@@ -152,16 +152,13 @@
 #define CF_AD9361_RX_DMA_BASEADDR	2
 #define CF_AD9361_TX_DMA_BASEADDR	3
 
-#define SPI_DEVICE_ID		0
+#define SPI_DEVICE_ID		1
```
Comment (dbogdan) [inline projects/ad9361/src/parameters.h:155]: The SPI_DEVICE_ID change is unrelated to this commit.
By the way, was this change needed for your setup? What .dts did you use?

**Comment by dbogdan** [comment]
Other than the comment regarding SPI_DEVICE_ID, your series of patches looks good, so, from my point of view, it can be merged.
@kylex2 , thanks for this.

File: projects/ad9361/src/parameters.h (hunk 1 above)
Comment (kylex2) [reply projects/ad9361/src/parameters.h:155]: Thank you for your comment. You are right, it's a mistake. I will change it.

Device tree was generated with Xilinx tools from fmcomms2 project in [analogdevicesinc/hdl](https://github.com/analogdevicesinc/hdl/tree/master/projects/fmcomms2)

I havent tried but I guess it works as expected if you use device tree from [analogdevicesinc/linux](https://github.com/analogdevicesinc/linux/) e.g. zynq-zed-adv7511-ad9361-4-fmcomms2-3-4-userspace.dts

**Review by dbogdan — APPROVED** [review]

**Comment by buha** [comment]
looks good to me too, but please make sure to force push the change @dbogdan mentioned above

**Comment by dbogdan** [comment]
I guess it's not something related to your changes - I've triggered a new check.

**Comment by dbogdan** [comment]
Everything seems to be fine now, so these changes can be merged.
Thanks again.
